src/MassTransit.AmazonSqsTransport.Tests/AmazonSqsAddress_Specs.cs

```diff
@@ -0,0 +1,136 @@
+namespace MassTransit.AmazonSqsTransport.Tests
+{
+    using System;
+    using NUnit.Framework;
+
+
+    [TestFixture]
+    public class Given_a_valid_host_address
+    {
+        [Test]
+        public void Should_return_a_valid_address_for_a_full_address_with_scope()
+        {
+            var host = new Uri("amazonsqs://remote-host/production");
+            var address = new AmazonSqsHostAddress(host);
+
+            Assert.That(address.Scope, Is.EqualTo("production"));
+            Assert.That(address.Host, Is.EqualTo("remote-host"));
+
+            Uri uri = address;
+
+            Assert.That(uri, Is.EqualTo(host));
+        }
+
+        [Test]
+        public void Should_return_a_valid_address_for_a_full_address()
+        {
+            var host = new Uri("amazonsqs://remote-host");
+            var address = new AmazonSqsHostAddress(host);
+
+            Assert.That(address.Scope, Is.EqualTo("/"));
+            Assert.That(address.Host, Is.EqualTo("remote-host"));
+
+            Uri uri = address;
+
+            Assert.That(uri, Is.EqualTo(host));
+        }
+    }
+
+
+    [TestFixture]
+    public class Given_a_valid_endpoint_address
+    {
+        [Test]
+        public void Should_return_a_valid_address_for_a_queue()
+        {
+            var hostAddress = new Uri("amazonsqs://docker.localhost/test");
+
+            var address = new AmazonSqsEndpointAddress(hostAddress, new Uri("queue:input-queue"));
+
+            Uri uri = address;
+
+            Assert.That(uri, Is.EqualTo(new Uri("amazonsqs://docker.localhost/test/input-queue")));
+        }
+
+        [Test]
+        public void Should_return_a_valid_address_for_a_full_address_with_scope()
```

Comment:
**So, don't change anything, I've already added my own things on top of your commit.**

I disagree with this test, I think if the host address specified has a _test_ scope, that it should override the scope specified in the endpoint address. I'm likely going to change it to behave that way, and update the test.

Reply:
tbh, it was copied from RMQ as well (but changed to Scope instead VirtualHost)

Reply:
Yeah, that's my concern. I guess I'll leave it for now to avoid breaking things, but I really think this needs revisited.